massiv/src/Data/Massiv/Core/List.hs

```diff
@@ -36,11 +37,16 @@ import           Data.Massiv.Core.Common
 import           Data.Massiv.Core.Scheduler
 import           Data.Proxy
 import           Data.Typeable
+import           Data.Validity
 import           GHC.Exts
+import           GHC.Generics (Generic)
 import           System.IO.Unsafe           (unsafePerformIO)
 
 data LN
 
+instance Validity LN where
+    validate = trivialValidation
```

Comment:
This _should_ be generatable. It doesn't really matter because you can never call it anyway, but best not to make extra code if we can avoid it.

Reply:
You're not actually saving code by generating it. You need to add DeriveGeneric, GHC.Generics and "deriving Generic" (for LN) before you can derive it.

Reply:
Same comment as above, up to @lehins